sdk/python/feast/feature_store.py

```diff
@@ -597,24 +621,40 @@ def _group_refs(
     # view name to feature names
     views_features = defaultdict(list)
 
+    feature_set = set()
+    feature_collision_set = set()
```

Comment:
Is there a reason why validation is mixed into `group_refs()` and not a standalone function?

Reply:
The logic in `_group_refs` lends itself well to checking for feature name collisions, but more importantly I think placing the check here is reasonable considering the description of the `_group_refs` function: """ Get list of feature views and corresponding feature names based on feature references""".

If the feature names are invalid (e.g. have a collision) this would be a fitting place to throw the error, I believe. Do let me know if you disagree.

Reply:
`_group_refs` is simply not descriptive since its not meant for validation and doesn't hint at what you are doing. This can be seen in your tests when you call `_group_refs` but its not clear why you are calling it.

It seems like the only benefit to using this method is that your a looping over the names once.

Can we at least change the name to something like `_validate_and_group_feature_refs`?